townhall/myapi/views.py

```diff
@@ -177,6 +177,57 @@ def get_all_opportunities_of_a_volunteer_request(self, request, vol_id=None):
             # If services method returns an error, return an error Response
             return Response({"message": str(e)}, status=status.HTTP_404_NOT_FOUND)
 
+    # DELETE A Volunteer
+    @action(detail=True, methods=["delete"], url_path="volunteer")
+    def delete_volunteer_request(self, request, vol_id=None):
+        # Get the volunteer id from the url
+        volunteer_id = vol_id
+
+        try:
+            # Call the service method to delete the volunteer
+            volunteer_services.delete_volunteer(volunteer_id)
```

Comment:
This method in the type definition expectes an int but youre passing in an int or None (if someone did something silly and passed in a bad volunteer_id) Its good you have the try catch for all other errors but before the try block you can already prevent this by setting up a guard. 
if volunteer_id is None:
      return 404